actgen/agents/dqn.py

```diff
@@ -99,6 +109,18 @@ def _get_action_similarities(self, batch):
             for i, acted in enumerate(action_taken):
                 action_similarity_scores = self.get_duplicate_actions_fn(acted)
                 similarity_mat[i] = torch.as_tensor(action_similarity_scores)  # update each row in similarity_mat
+        elif self.params['inv_model']:
+            # don't use the inverse model in early stages of training
+            if self.n_training_steps < self.params['inv_warmup_steps']:
+                return similarity_mat
+            # use the inverse model
+            action_probs = self.inverse_predictor.predict(batch, encoder=self.q.encoder)  # (batch_size, n_actions)
+            # scale the probabilities linearly so that the actual action taken has probability 1
+            scale = 1 / extract(action_probs, action_taken, idx_dim=-1).view((self.params['batch_size'], 1))
+            scaled_action_probs = scale * action_probs
+            # those with probabilities lower than 0.5 are deemed none-similar actions
+            clipped_action_probs = torch.where(scaled_action_probs < self.params['inv_clip_threshold'], 0, scaled_action_probs)
```

Comment:
If this doesn't work, let's try without rescaling and clipping. Instead we just modify the acted action similarity to equal 1.0.